samples/RhythmGame/src/main/cpp/audio/NDKExtractor.cpp

```diff
@@ -154,23 +154,22 @@ int32_t NDKExtractor::decode(AAsset *asset, uint8_t *targetData, AudioProperties
                 if (info.flags & AMEDIACODEC_BUFFER_FLAG_END_OF_STREAM){
                     LOGD("Reached end of decoding stream");
                     isDecoding = false;
-                } else {
-                    // Valid index, acquire buffer
-                    size_t outputSize;
-                    uint8_t *outputBuffer = AMediaCodec_getOutputBuffer(codec, outputIndex, &outputSize);
-
-                    /*LOGV("Got output buffer index %d, buffer size: %d, info size: %d writing to pcm index %d",
-                         outputIndex,
-                         outputSize,
-                         info.size,
-                         m_writeIndex);*/
-
-                    // copy the data out of the buffer
-                    memcpy(targetData + bytesWritten, outputBuffer, info.size);
-                    bytesWritten+=info.size;
-                    AMediaCodec_releaseOutputBuffer(codec, outputIndex, false);
-                }
-
+                } 
+                
+                // Valid index, acquire buffer
+                size_t outputSize;
+                uint8_t *outputBuffer = AMediaCodec_getOutputBuffer(codec, outputIndex, &outputSize);
+
+                /*LOGV("Got output buffer index %d, buffer size: %d, info size: %d writing to pcm index %d",
+                     outputIndex,
+                     outputSize,
+                     info.size,
+                     m_writeIndex);*/
+
+                // copy the data out of the buffer
+                memcpy(targetData + bytesWritten, outputBuffer, info.size);
+                bytesWritten+=info.size;
+                AMediaCodec_releaseOutputBuffer(codec, outputIndex, false);
```

Comment:
The following code needs to be executed also if end of stream has been reached. Otherwise last samples are not decoded.

Reply:
Good spot - thanks.